libass/ass_cache.c

```diff
@@ -373,6 +420,14 @@ void *ass_cache_get(Cache *cache, void *key, void *priv)
             cache->hits++;
             desc->key_move_func(NULL, key);
             item->ref_count++;
+
+#if CONFIG_PTHREAD
+            while (!item->size)
+                pthread_cond_wait(&cache->cond, &cache->mutex);
```

Comment:
I think, my original idea was to use per-thread convar here to minimize interference with other items. Dunno how large performance impact can be though.

Reply:
Per-thread or per-bucket?

Reply:
Per thread: `ass_cache_get()` takes an additional parameter, stores it in `CacheItem` during `construct_func()`, and other threads wait it. I recall, it was not even a condvar, I planned to use per-thread mutex in conjunction with `ref_count` atomics.

Reply:
Hmm, in my local testing I'm spending a negligible amount of time in this particular condvar wait (<0.1%), but that could be interesting to explore if it turns out to be a bigger deal on other samples. (I'm spending far more time blocked on the cache mutex, or in mutex overhead when locking and unlocking.)